### PR TITLE
chore: update eslint + prettier

### PR DESCRIPTION
**Pull request by SamWooler**
**Changes**
- Bumped Prettier to v2.3.2
- Bumped ESLint to v7.0.0

**ESLint**
Version 7 of ESLint has deprecated the following rules.

- callback-return
- global-require
- handle-callback-err
- no-buffer-constructor
- no-mixed-requires
- no-new-require
- no-path-concat
- no-process-env
- no-process-exit
- no-restricted-modules

These have now been added into `eslint-plugin-node`. This has been added into the config, old rules have been ignored and the new rules set according to whether they are errors (marked as error) or stylistic (marked as off). The package [eslint-plugin-node](https://github.com/mysticatea/eslint-plugin-node) details which are errors and which are stylistic.

### CODE REVIEW

File: packages/eslint-config-datacamp/index.js [hunk 1]
```diff
@@ -73,8 +81,27 @@ module.exports = {
     'jest/no-deprecated-functions': 'off', // Needs to know the jest version, not possible from a shared config
     'max-classes-per-file': 'off',
     'no-console': 'error',
+    'no-mixed-requires': 'off', // Deprecated in ESLint 7.0.0, uses node/no-mixed-requires instead
```
Comment (blurbyte) [inline packages/eslint-config-datacamp/index.js:84]: I am wondering what is the best practice in such case: is it better to do it like here - turn it off, or rather remove it completely from config.

File: prettier.config.js [hunk 2]
```diff
@@ -1,3 +1,3 @@
 module.exports = {
-  ...require('@datacamp/prettier-config'),
+  ...require('./packages/prettier-config-datacamp'),
```
Comment (blurbyte) [inline prettier.config.js:2]: Linter was complaining here, so provided direct path to config.

**Review by blurbyte — COMMENTED** [review]
Looks good! I am wondering if we have to turn off deprecated rules explicitly or they are off by default (according to docs you can still use deprecated rules, but I guess you have to turn them on).
Probably we need more feedback on this.

File: packages/eslint-config-datacamp/index.js (hunk 1 above)
Comment (entropitor) [reply packages/eslint-config-datacamp/index.js:84]: @blurbyte The thing is that we added a plugin that replaces these rules. We turned them off so they are not double reported

Comment (blurbyte) [reply packages/eslint-config-datacamp/index.js:84]: Yeah, probably some presets may have them turned on explicitly (and not updated to eslint 7) so I guess it's better to turn them off.

**Pull request merged**